### PR TITLE
fix(app): set per-page document titles (#2295)

### LINKED ISSUE #2295: Set a per-page HTML title tag so browser history is distinguishable
### What problem are you trying to solve?

Every page in the app renders the same document title, `LFX`. It is set once in `apps/lfx-one/src/index.html` and, apart from the public profile and docs pages, no route ever changes it.

The user-visible consequence is in the browser's back-button menu: after visiting several pages, the history list is a column of identical "LFX" entries, so there is no way to pick the page you want to return to. The same undifferentiated title also shows in browser tabs, bookmarks, and window switchers.

A secondary effect: the analytics services read `document.title` when reporting page views, so those events currently carry no page identity either.

### What's your suggested solution?

Give each route a meaningful document title that is applied on navigation, following the `<Page> · LFX` shape already used by the public profile page.

Angular has first-class support for this — a route-level `title` key combined with a custom `TitleStrategy` — so route definitions can declare their title declaratively rather than each component setting it imperatively. Pages whose title depends on loaded data (a specific project, committee, meeting, or profile) need the resolved name folded in once that data arrives.

Note that many routes already carry a `data.title` value used for page headers and navigation; that is a good source for the wording, but it is not the key Angular's title feature reads, so it does not affect `<title>` today.

### Examples (if relevant)

- Current: every entry in the back menu reads `LFX`.
- Desired: `Project Meetings · LFX`, `Org Overview · LFX`, `Settings · LFX`, `Kubernetes · LFX`.
- Existing precedent in the codebase:
  - `apps/lfx-one/src/app/modules/public-profile/public-profile-page/public-profile-page.component.ts`
  - `apps/lfx-one/src/app/modules/docs/pages/docs-article/docs-article.component.ts`

### Definition of Done

- Navigating between pages produces distinct, human-readable entries in the browser's back and forward history menus.
- Every route reachable from the main navigation has a title; no page falls back to a bare `LFX`.
- Titles for data-driven pages include the relevant entity name once it has loaded.
- Titles are applied under SSR as well as client-side navigation.
- Analytics page-view events carry the page-specific title.

### PR DESCRIPTION
**Pull request by MRashad26**
## Summary
- Apply a distinct `{Page} · LFX` document title on every navigation (SSR and client) via a custom `TitleStrategy`, so browser history, tabs, and analytics are no longer a column of identical `LFX` entries.
- Declare titles on main-nav and feature routes (reusing existing `data.title` where present) and fold loaded entity names into project, group, meeting, mailing-list, and profile pages.
- Keep docs titles as `{Article} · LFX Documentation` without double-suffixing.

Fixes #2295

## Test plan
- [ ] Navigate across Me / Foundation / Project / Org pages and confirm tab titles follow `{Page} · LFX` (e.g. `Project Meetings · LFX`, `Org Overview · LFX`, `Settings · LFX`).
- [ ] Open the browser back-button menu after several navigations and confirm entries are distinguishable.
- [ ] Open a project dashboard and confirm the title becomes `{Project name} · LFX` once the project loads.
- [ ] Open a meeting join page, group detail, committee, mailing list, and public profile; confirm the entity name appears in the title after load.
- [ ] Open a docs article and confirm the title stays `{Article} · LFX Documentation` (not `… · LFX · LFX`).
- [ ] Hard-refresh a titled route and confirm SSR HTML includes the page-specific `<title>`.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: docs/architecture/frontend/angular-patterns.md [hunk 1]
```diff
@@ -242,6 +242,29 @@ export class ExampleComponent {
...
 ```
 
+## Document titles
+
+Every navigable route gets a distinct `document.title` so browser history, tabs, and analytics can tell pages apart. Titles follow the `{Page} · LFX` shape (the public profile page's existing convention). Docs articles keep `{Article} · LFX Documentation`.
+
+Angular applies titles through a custom `TitleStrategy`:
+
+1. Declare `title: 'My Meetings'` on the route (first-class Angular key), **or** reuse an existing `data.title` used for page headers.
+2. `LfxTitleStrategy` (provided in `app.config.ts`) reads the deepest primary-outlet title on every navigation — including SSR — and suffixes ` · LFX` unless the value is already branded.
```
Comment (audigregorie) [inline docs/architecture/frontend/angular-patterns.md:252]: **Critical — `markdown-lint` fails on this line (MD038), CI is red on the PR head**

The `` ` · LFX` `` code span starts with a space, which violates MD038/no-space-in-code. The `markdown-lint` workflow fails on the current head with:

```
docs/architecture/frontend/angular-patterns.md:252:143 MD038/no-space-in-code Spaces inside code span elements [Context: "` · LFX`"]
```

Reword so the code span doesn't begin with a space:

```suggestion
2. `LfxTitleStrategy` (provided in `app.config.ts`) reads the deepest primary-outlet title on every navigation — including SSR — and appends the `· LFX` brand suffix unless the value is already branded.
```

File: apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * value is already branded (docs / public profile).
+ *
+ * Leaves untitled routes at the brand (`LFX`) rather than keeping a previous page's title.
+ */
+@Injectable({ providedIn: 'root' })
+export class LfxTitleStrategy extends TitleStrategy {
+  private readonly title = inject(Title);
+
+  public override updateTitle(routerState: RouterStateSnapshot): void {
+    this.title.setTitle(formatLfxDocumentTitle(this.resolvePageTitle(routerState)));
```
Comment (audigregorie) [inline apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts:24]: **Major — `updateTitle` runs on every `NavigationEnd`, so entity titles are clobbered by same-route navigations and never restored**

Angular calls `TitleStrategy.updateTitle` after *every* successful navigation, including same-route query-param-only updates (verified in `@angular/router@20.3.15`: `updateTitle` fires immediately after each `NavigationEnd`). `bindLfxDocumentTitle` only re-emits when the entity name *changes*, so once this strategy re-applies the static route title, the entity name is gone until the entity itself reloads.

Concrete repro on a page this PR instruments: open `/org/projects/:projectSlug`, wait for the title to become `{Project} · LFX`, then switch tabs (or metric / time range). `OrgProjectDetailComponent.switchTab` / `setMetric` / `setTimeRange` all do `router.navigate([], { relativeTo: this.route, queryParams: …, queryParamsHandling: 'merge', replaceUrl: true })` → `updateTitle` resets the tab title to `Project Detail · LFX`, `hero()?.projectName` hasn't changed so nothing re-emits, and the project name never comes back. The same exposure exists for the `?step=` manage wizards and any future filter-in-URL navigation on the meeting/group/committee/mailing-list pages.

Suggested direction (strategy-side, so it covers every current and future binder): remember the previous activation's primary-outlet route-config chain, and when it's unchanged (pure query/param navigation on the same component tree) and `Title.getTitle()` no longer matches what the strategy last set, skip the reset — a component-level title (`bindLfxDocumentTitle`, docs articles, the public profile) owns it:

```ts
private lastChainKey: string | undefined;
private lastAppliedRouteTitle: string | undefined;

public override updateTitle(routerState: RouterStateSnapshot): void {
  const chainKey = this.primaryConfigChain(routerState.root); // e.g. configs joined by '/'
  const sameChain = chainKey === this.lastChainKey;
  this.lastChainKey = chainKey;

  const next = formatLfxDocumentTitle(this.resolvePageTitle(routerState));
  // Same component tree, and something else (entity binding / docs article) has taken over the
  // title since the last route-level application — keep it.
  if (sameChain && this.lastAppliedRouteTitle !== undefined && this.title.getTitle() !== this.lastAppliedRouteTitle) {
    return;
  }
  this.lastAppliedRouteTitle = next;
  this.title.setTitle(next);
}
```

Exact heuristic is up to you — the important property is that a same-config-chain navigation must not stomp a component-applied title.

File: apps/lfx-one/src/app/modules/docs/docs.routes.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export const DOCS_ROUTES: Routes = [
   },
   {
     path: 'not-found',
+    title: 'Page not found · LFX Documentation',
```
Comment (audigregorie) [inline apps/lfx-one/src/app/modules/docs/docs.routes.ts:33]: **Minor — hardcoded brand segment duplicates `LFX_DOCUMENT_TITLE_DOCS_BRAND`**

This is the only route title that inlines the brand. It works today because `formatLfxDocumentTitle` exempts strings ending in ` · LFX Documentation`, but it hardcodes that exact value — if `LFX_DOCUMENT_TITLE_DOCS_BRAND` ever changes, this silently drifts and the double-suffix exemption stops matching (producing `… · LFX Documentation · LFX`).

```suggestion
    title: `Page not found · ${LFX_DOCUMENT_TITLE_DOCS_BRAND}`,
```

(plus `import { LFX_DOCUMENT_TITLE_DOCS_BRAND } from '@lfx-one/shared/constants';` at the top of the file)

File: apps/lfx-one/src/app/app.routes.ts [hunk 4]
```diff
@@ -609,6 +676,7 @@ export const routes: Routes = [
     children: [
       {
         path: '**',
+        title: 'Page not found',
```
Comment (audigregorie) [inline apps/lfx-one/src/app/app.routes.ts:679]: **Minor — casing is inconsistent with the other not-found titles added in this PR**

`Page not found` (sentence case) here and in `docs.routes.ts`, but `Meeting Not Found` / `Group Not Found` (title case) a few lines up in this file. Worth picking one convention so history/tab entries look uniform.

**Review by audigregorie — COMMENTED** [review]
## Code Review Summary

Thorough, well-scoped change — full route coverage, shared constants, unit tests for both the formatter and the strategy, and architecture docs updated in the same PR. Two things need attention before merge: the `markdown-lint` CI check is red on the current head (MD038 in `angular-patterns.md`), and entity titles applied via `bindLfxDocumentTitle` are reset by the `TitleStrategy` hook on *every* navigation — including same-route query-param changes — so they regress on first interaction (e.g. org project-detail tab switch) and never recover.

### Outside the diff

- The PR description's test plan covers load-time titles well; worth adding a "switch tabs / filters on an entity page and confirm the title survives" case once the `updateTitle` clobbering is addressed.

### What's done well

- Reuses the `*_LABEL` constants (`COMMITTEE_LABEL`, `MAILING_LIST_LABEL`, …) so titles track product renames instead of hardcoding strings.
- `bindLfxDocumentTitle` follows the repo's `toObservable` + `takeUntilDestroyed` convention instead of `effect()` (frontend-checklist §5), and empty emissions are deliberately ignored so the route title holds until data arrives.
- The org-lens `data.title` fallback is a clean way to avoid duplicating page-header copy.
- Formatter tests cover the double-suffix and docs-brand edge cases; strategy tests cover parent/leaf precedence and the data fallback.

File: docs/architecture/frontend/angular-patterns.md (hunk 1 above)
Comment (MRashad26) [reply docs/architecture/frontend/angular-patterns.md]: Done — reworded the code span to `· LFX` (no leading space) so MD038 is clean.

See commit 1e176f90b.

File: apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts (hunk 2 above)
Comment (MRashad26) [reply apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts]: Done — `LfxTitleStrategy` now skips the reset when the primary-outlet config chain is unchanged and a component already owns the title (tab/filter/`?step=` navigations). Covered by a strategy spec that sets an entity title then changes query params.

See commit 1e176f90b.

File: apps/lfx-one/src/app/shared/utils/document-title.util.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ */
+export function bindLfxDocumentTitle(page: Signal<string | null | undefined>): void {
+  const title = inject(Title);
+  const destroyRef = inject(DestroyRef);
+  toObservable(page)
+    .pipe(takeUntilDestroyed(destroyRef))
+    .subscribe((value) => {
+      const next = value?.trim();
+      if (next) {
+        title.setTitle(formatLfxDocumentTitle(next));
```
[the remark this answers is not in the document]
Comment (MRashad26) [reply apps/lfx-one/src/app/shared/utils/document-title.util.ts:25]: Fixed in `LfxTitleStrategy` rather than re-emitting from this binder, so every current and future `bindLfxDocumentTitle` / `Title.setTitle` caller is covered. Same-route query/param navigations no longer clobber the entity title.

See commit 1e176f90b.

File: apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  }
```
[the remark this answers is not in the document]
Comment (MRashad26) [reply apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts:59]: Done — Plausible and Segment now read `document.title` on a microtask after `NavigationEnd` (`afterRouterTitleApplied`), which is after `TitleStrategy.updateTitle`. Added a strategy spec that asserts navigation A→B still shows A during `NavigationEnd` and B on the following microtask.

See commit 1e176f90b.

File: apps/lfx-one/src/app/modules/newsletters/newsletters.routes.ts [hunk 7]
```diff
@@ -13,12 +13,14 @@ export const NEWSLETTER_ROUTES: Routes = [
   },
   {
     path: 'list',
+    title: 'Newsletters',
     canActivate: [authGuard, newsletterAccessGuard],
     loadComponent: () => import('./newsletter-list/newsletter-list.component').then((m) => m.NewsletterListComponent),
```
[the remark this answers is not in the document]
Comment (MRashad26) [reply apps/lfx-one/src/app/modules/newsletters/newsletters.routes.ts:19]: Done — removed the list-route `title` so `/foundation/newsletters` and `/project/newsletters` inherit `Foundation Newsletters` / `Project Newsletters`. Create/my/edit/analytics keep their own titles.

See commit 1e176f90b.

File: apps/lfx-one/src/app/modules/docs/docs.routes.ts (hunk 3 above)
Comment (MRashad26) [reply apps/lfx-one/src/app/modules/docs/docs.routes.ts]: Done — the docs 404 route (and the not-found component, so they stay in sync) now builds the title from `LFX_DOCUMENT_TITLE_SEPARATOR` + `LFX_DOCUMENT_TITLE_DOCS_BRAND`.

See commit 1e176f90b.

File: apps/lfx-one/src/app/app.routes.ts (hunk 4 above)
Comment (MRashad26) [reply apps/lfx-one/src/app/app.routes.ts]: Done — aligned on title case with `Meeting Not Found` / `Group Not Found`: document titles are now `Page Not Found` (and `Page Not Found · LFX Documentation` on docs). The in-page H1 copy is unchanged.

See commit 1e176f90b.

**Comment by MRashad26** [comment]
## Review Feedback Addressed

Commit: `1e176f90b1ae279c19bbb59af668122beda68da2`

### Changes Made
- **`docs/architecture/frontend/angular-patterns.md`**: reworded the brand suffix so MD038 (`no-space-in-code`) no longer fails CI (per @audigregorie)
- **`LfxTitleStrategy`**: same-route query/param navigations no longer overwrite a component-applied entity title (per @audigregorie, Copilot)
- **`PlausibleService` / `SegmentService`**: read `document.title` after `TitleStrategy.updateTitle` via a microtask (per Copilot)
- **`newsletters.routes.ts`**: list route inherits the mount title so Foundation/Project/Me history entries stay distinct (per Copilot)
- **`docs.routes.ts`**: docs 404 title uses `LFX_DOCUMENT_TITLE_DOCS_BRAND` (per @audigregorie)
- **`app.routes.ts`**: not-found document titles use title case (`Page Not Found`) to match `Meeting Not Found` / `Group Not Found` (per @audigregorie)

### Threads Resolved
7 of 7 unresolved threads addressed in this iteration.

File: apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  private primaryConfigChain(root: ActivatedRouteSnapshot): string {
+    const parts: string[] = [];
+    let route: ActivatedRouteSnapshot | undefined = root;
+    while (route) {
+      parts.push(route.routeConfig?.path ?? '');
+      route = route.children.find((child) => child.outlet === PRIMARY_OUTLET);
+    }
+    return parts.join('/');
+  }
```
[the remark this answers is not in the document]
Comment (MRashad26) [reply apps/lfx-one/src/app/shared/strategies/lfx-title.strategy.ts:59]: Done — the chain key now includes consumed URL segments (not query params), so `/projects/k8s` → `/projects/nodejs` resets to the route title while `?tab=` still keeps the loaded entity name. Regression is in the strategy spec.

See commit a49f4a676.

**Comment by MRashad26** [comment]
## Follow-up Review Feedback

Commit: `a49f4a676`

- **`LfxTitleStrategy`**: chain key now includes consumed path segments, so `/projects/k8s` → `/projects/nodejs` resets the title (analytics and error-state binders no longer keep the previous entity). Query-only navigations still preserve the loaded name (per Copilot).

1 of 1 remaining unresolved thread addressed.

**Review by audigregorie — APPROVED** [review]
